Engine/DataFeeds/TimeSlice.cs

```diff
@@ -158,6 +158,12 @@ public class TimeSlice
             // ensure we read equity data before option data, so we can set the current underlying price
             foreach (var packet in data)
             {
+                // filter out packets for removed subscriptions
+                if (packet.IsSubscriptionRemoved)
```

Comment:
Does this ever get hit now?

Reply:
This gets hit during universe selection algorithms. The algorithm listed above on current `master` has some time steps that emit data for 60 securities instead of 50. After this change, each time step emits data for 50 securities as expected. The underlying cause was enumerating the data feed subscription before performing universe selection. I had considered using sorting to enforce an order to the enumeration but felt this required less coupling/coordination between data feed and universe selection mechanics.